### PR TITLE
warn about potentially missing patches in --new-pr

### PR DESCRIPTION
**Pull request by migueldiascosta**
requested in https://github.com/easybuilders/easybuild-framework/issues/1674, at least

not sure if it always makes sense to automatically add patches, but we can at least warn about potentially missing ones...

### CODE REVIEW

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
What about --new-pr on easyconfig that uses patch files that already exist in the repo?
We don't want to warn about them do we?

**Comment by migueldiascosta** [comment]
> What about --new-pr on easyconfig that uses patch files that already exist in the repo? We don't want to warn about them do we?

right, addressed in https://github.com/easybuilders/easybuild-framework/commit/799c3b4e96d1f86353c1faab760a440912a47e87

File: easybuild/tools/github.py [hunk 1]
```diff
@@ -1783,6 +1783,11 @@ def new_pr(paths, ecs, title=None, descr=None, commit_msg=None):
     res = new_branch_github(paths, ecs, commit_msg=commit_msg)
     file_info, deleted_paths, _, branch_name, diff_stat, pr_target_repo = res
 
+    for ec, ec_path in zip(file_info['ecs'], file_info['paths_in_repo']):
+        for patch in ec.asdict()['patches']:
+            if patch not in paths['patch_files'] and not os.path.isfile(os.path.join(os.path.dirname(ec_path), patch)):
+                print_warning("new patch %s, referenced by %s, is not included in this PR" % (patch, ec.filename()))
```
Comment (akesandgren) [inline easybuild/tools/github.py:1789]: perhaps "new patch file"?? but do as you please.

**Comment by akesandgren** [comment]
All in all, this looks good and it's something I could have benefited from multiple times... ooohh so many times...

**Comment by akesandgren** [comment]
oops hit the wrong button

**Comment by akesandgren** [comment]
Change to non-draft?

**Review by akesandgren — DISMISSED** [review]
LGTM

**Comment by migueldiascosta** [comment]
@akesandgren wait, the tests requiring github token need to be updated

**Comment by akesandgren** [comment]
I have one more question, is the file_info['ecs'] and file_info['paths_in_repo'] guaranteed to match in number of elements?
Haven't had time to check the code yet...

**Comment by akesandgren** [comment]
Have you run the framework test suite after this commit?

**Comment by migueldiascosta** [comment]
> Have you run the framework test suite after this commit?

I ran the (50) tests that require the github token locally, yes, they all pass now

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @migueldiascosta!

**Pull request merged**